{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Downloading/unpacking scipy\n",
      "Downloading/unpacking scipy\n",
      "  Running setup.py (path:/tmp/pip_build_root/scipy/setup.py) egg_info for package scipy\n",
      "  Running setup.py (path:/tmp/pip_build_root/scipy/setup.py) egg_info for package scipy\n",
      "    Running from numpy source directory.\n",
      "    Running from numpy source directory.\n",
      "    /tmp/easy_install-3_H80f/numpy-1.11.2rc1/setup.py:327: UserWarning: Unrecognized setuptools command, proceeding with generating Cython sources and expanding templates\n",
      "    /tmp/easy_install-3_H80f/numpy-1.11.2rc1/setup.py:327: UserWarning: Unrecognized setuptools command, proceeding with generating Cython sources and expanding templates\n",
      "      warnings.warn(\"Unrecognized setuptools command, proceeding with \"\n",
      "      warnings.warn(\"Unrecognized setuptools command, proceeding with \"\n",
      "    /tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/system_info.py:1630: UserWarning:\n",
      "    /tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/system_info.py:1630: UserWarning:\n",
      "        Atlas (http://math-atlas.sourceforge.net/) libraries not found.\n",
      "        Atlas (http://math-atlas.sourceforge.net/) libraries not found.\n",
      "        Directories to search for the libraries can be specified in the\n",
      "        Directories to search for the libraries can be specified in the\n",
      "        numpy/distutils/site.cfg file (section [atlas]) or by setting\n",
      "        numpy/distutils/site.cfg file (section [atlas]) or by setting\n",
      "        the ATLAS environment variable.\n",
      "        the ATLAS environment variable.\n",
      "      warnings.warn(AtlasNotFoundError.__doc__)\n",
      "      warnings.warn(AtlasNotFoundError.__doc__)\n",
      "    /usr/bin/ld: cannot find -lcblas\n",
      "    /usr/bin/ld: cannot find -lcblas\n",
      "    collect2: error: ld returned 1 exit status\n",
      "    collect2: error: ld returned 1 exit status\n",
      "    /bin/sh: 1: svnversion: not found\n",
      "    /bin/sh: 1: svnversion: not found\n",
      "    /bin/sh: 1: svnversion: not found\n",
      "    /bin/sh: 1: svnversion: not found\n",
      "    /tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/system_info.py:1532: UserWarning:\n",
      "    /tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/system_info.py:1532: UserWarning:\n",
      "        Atlas (http://math-atlas.sourceforge.net/) libraries not found.\n",
      "        Atlas (http://math-atlas.sourceforge.net/) libraries not found.\n",
      "        Directories to search for the libraries can be specified in the\n",
      "        Directories to search for the libraries can be specified in the\n",
      "        numpy/distutils/site.cfg file (section [atlas]) or by setting\n",
      "        numpy/distutils/site.cfg file (section [atlas]) or by setting\n",
      "        the ATLAS environment variable.\n",
      "        the ATLAS environment variable.\n",
      "      warnings.warn(AtlasNotFoundError.__doc__)\n",
      "      warnings.warn(AtlasNotFoundError.__doc__)\n",
      "    /usr/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'define_macros'\n",
      "    /usr/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'define_macros'\n",
      "      warnings.warn(msg)\n",
      "      warnings.warn(msg)\n",
      "    _configtest.c:1:5: warning: conflicting types for built-in function ‘exp’ [enabled by default]\n",
      "    _configtest.c:1:5: warning: conflicting types for built-in function ‘exp’ [enabled by default]\n",
      "     int exp (void);\n",
      "     int exp (void);\n",
      "         ^\n",
      "         ^\n",
      "    _configtest.o: In function `main':\n",
      "    _configtest.o: In function `main':\n",
      "    /tmp/easy_install-3_H80f/numpy-1.11.2rc1/_configtest.c:6: undefined reference to `exp'\n",
      "    /tmp/easy_install-3_H80f/numpy-1.11.2rc1/_configtest.c:6: undefined reference to `exp'\n",
      "    collect2: error: ld returned 1 exit status\n",
      "    collect2: error: ld returned 1 exit status\n",
      "    _configtest.c:1:5: warning: conflicting types for built-in function ‘exp’ [enabled by default]\n",
      "    _configtest.c:1:5: warning: conflicting types for built-in function ‘exp’ [enabled by default]\n",
      "     int exp (void);\n",
      "     int exp (void);\n",
      "         ^\n",
      "         ^\n",
      "    _configtest.c:1:20: fatal error: Python.h: No such file or directory\n",
      "    _configtest.c:1:20: fatal error: Python.h: No such file or directory\n",
      "     #include <Python.h>\n",
      "     #include <Python.h>\n",
      "                        ^\n",
      "                        ^\n",
      "    compilation terminated.\n",
      "    compilation terminated.\n",
      "    /usr/bin/ld: cannot find -lcblas\n",
      "    /usr/bin/ld: cannot find -lcblas\n",
      "    collect2: error: ld returned 1 exit status\n",
      "    collect2: error: ld returned 1 exit status\n",
      "    non-existing path in 'numpy/distutils': 'site.cfg'\n",
      "    non-existing path in 'numpy/distutils': 'site.cfg'\n",
      "    _configtest.o: In function `main':\n",
      "    _configtest.o: In function `main':\n",
      "    /tmp/easy_install-3_H80f/numpy-1.11.2rc1/_configtest.c:6: undefined reference to `exp'\n",
      "    /tmp/easy_install-3_H80f/numpy-1.11.2rc1/_configtest.c:6: undefined reference to `exp'\n",
      "    collect2: error: ld returned 1 exit status\n",
      "    collect2: error: ld returned 1 exit status\n",
      "    _configtest.c:1:20: fatal error: Python.h: No such file or directory\n",
      "    _configtest.c:1:20: fatal error: Python.h: No such file or directory\n",
      "     #include <Python.h>\n",
      "     #include <Python.h>\n",
      "                        ^\n",
      "                        ^\n",
      "    compilation terminated.\n",
      "    compilation terminated.\n",
      "    Traceback (most recent call last):\n",
      "    Traceback (most recent call last):\n",
      "      File \"<string>\", line 17, in <module>\n",
      "      File \"<string>\", line 17, in <module>\n",
      "      File \"/tmp/pip_build_root/scipy/setup.py\", line 415, in <module>\n",
      "      File \"/tmp/pip_build_root/scipy/setup.py\", line 415, in <module>\n",
      "        setup_package()\n",
      "        setup_package()\n",
      "      File \"/tmp/pip_build_root/scipy/setup.py\", line 411, in setup_package\n",
      "      File \"/tmp/pip_build_root/scipy/setup.py\", line 411, in setup_package\n",
      "        setup(**metadata)\n",
      "        setup(**metadata)\n",
      "      File \"/usr/lib/python2.7/distutils/core.py\", line 111, in setup\n",
      "      File \"/usr/lib/python2.7/distutils/core.py\", line 111, in setup\n",
      "        _setup_distribution = dist = klass(attrs)\n",
      "        _setup_distribution = dist = klass(attrs)\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/dist.py\", line 315, in __init__\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/dist.py\", line 315, in __init__\n",
      "        self.fetch_build_eggs(attrs['setup_requires'])\n",
      "        self.fetch_build_eggs(attrs['setup_requires'])\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/dist.py\", line 361, in fetch_build_eggs\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/dist.py\", line 361, in fetch_build_eggs\n",
      "        replace_conflicting=True,\n",
      "        replace_conflicting=True,\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py\", line 851, in resolve\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py\", line 851, in resolve\n",
      "        dist = best[req.key] = env.best_match(req, ws, installer)\n",
      "        dist = best[req.key] = env.best_match(req, ws, installer)\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py\", line 1123, in best_match\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py\", line 1123, in best_match\n",
      "        return self.obtain(req, installer)\n",
      "        return self.obtain(req, installer)\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py\", line 1135, in obtain\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py\", line 1135, in obtain\n",
      "        return installer(requirement)\n",
      "        return installer(requirement)\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/dist.py\", line 429, in fetch_build_egg\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/dist.py\", line 429, in fetch_build_egg\n",
      "        return cmd.easy_install(req)\n",
      "        return cmd.easy_install(req)\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/easy_install.py\", line 665, in easy_install\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/easy_install.py\", line 665, in easy_install\n",
      "        return self.install_item(spec, dist.location, tmpdir, deps)\n",
      "        return self.install_item(spec, dist.location, tmpdir, deps)\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/easy_install.py\", line 695, in install_item\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/easy_install.py\", line 695, in install_item\n",
      "        dists = self.install_eggs(spec, download, tmpdir)\n",
      "        dists = self.install_eggs(spec, download, tmpdir)\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/easy_install.py\", line 876, in install_eggs\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/easy_install.py\", line 876, in install_eggs\n",
      "        return self.build_and_install(setup_script, setup_base)\n",
      "        return self.build_and_install(setup_script, setup_base)\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/easy_install.py\", line 1115, in build_and_install\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/easy_install.py\", line 1115, in build_and_install\n",
      "        self.run_setup(setup_script, setup_base, args)\n",
      "        self.run_setup(setup_script, setup_base, args)\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/easy_install.py\", line 1101, in run_setup\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/easy_install.py\", line 1101, in run_setup\n",
      "        run_setup(setup_script, args)\n",
      "        run_setup(setup_script, args)\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 249, in run_setup\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 249, in run_setup\n",
      "        raise\n",
      "        raise\n",
      "      File \"/usr/lib/python2.7/contextlib.py\", line 35, in __exit__\n",
      "      File \"/usr/lib/python2.7/contextlib.py\", line 35, in __exit__\n",
      "        self.gen.throw(type, value, traceback)\n",
      "        self.gen.throw(type, value, traceback)\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 197, in setup_context\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 197, in setup_context\n",
      "        yield\n",
      "        yield\n",
      "      File \"/usr/lib/python2.7/contextlib.py\", line 35, in __exit__\n",
      "      File \"/usr/lib/python2.7/contextlib.py\", line 35, in __exit__\n",
      "        self.gen.throw(type, value, traceback)\n",
      "        self.gen.throw(type, value, traceback)\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 168, in save_modules\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 168, in save_modules\n",
      "        saved_exc.resume()\n",
      "        saved_exc.resume()\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 143, in resume\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 143, in resume\n",
      "        six.reraise(type, exc, self._tb)\n",
      "        six.reraise(type, exc, self._tb)\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 156, in save_modules\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 156, in save_modules\n",
      "        yield saved\n",
      "        yield saved\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 197, in setup_context\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 197, in setup_context\n",
      "        yield\n",
      "        yield\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 246, in run_setup\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 246, in run_setup\n",
      "        DirectorySandbox(setup_dir).run(runner)\n",
      "        DirectorySandbox(setup_dir).run(runner)\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 276, in run\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 276, in run\n",
      "        return func()\n",
      "        return func()\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 245, in runner\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 245, in runner\n",
      "        _execfile(setup_script, ns)\n",
      "        _execfile(setup_script, ns)\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 47, in _execfile\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 47, in _execfile\n",
      "        exec(code, globals, locals)\n",
      "        exec(code, globals, locals)\n",
      "      File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/setup.py\", line 386, in <module>\n",
      "      File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/setup.py\", line 386, in <module>\n",
      "        # Raise errors for unsupported commands, improve help output, etc.\n",
      "        # Raise errors for unsupported commands, improve help output, etc.\n",
      "      File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/setup.py\", line 378, in setup_package\n",
      "      File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/setup.py\", line 378, in setup_package\n",
      "        test_suite='nose.collector',\n",
      "        test_suite='nose.collector',\n",
      "      File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/core.py\", line 169, in setup\n",
      "      File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/core.py\", line 169, in setup\n",
      "      File \"/usr/lib/python2.7/distutils/core.py\", line 151, in setup\n",
      "      File \"/usr/lib/python2.7/distutils/core.py\", line 151, in setup\n",
      "        dist.run_commands()\n",
      "        dist.run_commands()\n",
      "      File \"/usr/lib/python2.7/distutils/dist.py\", line 953, in run_commands\n",
      "      File \"/usr/lib/python2.7/distutils/dist.py\", line 953, in run_commands\n",
      "        self.run_command(cmd)\n",
      "        self.run_command(cmd)\n",
      "      File \"/usr/lib/python2.7/distutils/dist.py\", line 972, in run_command\n",
      "      File \"/usr/lib/python2.7/distutils/dist.py\", line 972, in run_command\n",
      "        cmd_obj.run()\n",
      "        cmd_obj.run()\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/bdist_egg.py\", line 152, in run\n",
      "      File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/bdist_egg.py\", line 152, in run\n",
      "        self.run_command(\"egg_info\")\n",
      "        self.run_command(\"egg_info\")\n",
      "      File \"/usr/lib/python2.7/distutils/cmd.py\", line 326, in run_command\n",
      "      File \"/usr/lib/python2.7/distutils/cmd.py\", line 326, in run_command\n",
      "        self.distribution.run_command(command)\n",
      "        self.distribution.run_command(command)\n",
      "      File \"/usr/lib/python2.7/distutils/dist.py\", line 972, in run_command\n",
      "      File \"/usr/lib/python2.7/distutils/dist.py\", line 972, in run_command\n",
      "        cmd_obj.run()\n",
      "        cmd_obj.run()\n",
      "      File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/command/egg_info.py\", line 18, in run\n",
      "      File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/command/egg_info.py\", line 18, in run\n",
      "      File \"/usr/lib/python2.7/distutils/cmd.py\", line 326, in run_command\n",
      "      File \"/usr/lib/python2.7/distutils/cmd.py\", line 326, in run_command\n",
      "        self.distribution.run_command(command)\n",
      "        self.distribution.run_command(command)\n",
      "      File \"/usr/lib/python2.7/distutils/dist.py\", line 972, in run_command\n",
      "      File \"/usr/lib/python2.7/distutils/dist.py\", line 972, in run_command\n",
      "        cmd_obj.run()\n",
      "        cmd_obj.run()\n",
      "      File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/command/build_src.py\", line 147, in run\n",
      "      File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/command/build_src.py\", line 147, in run\n",
      "      File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/command/build_src.py\", line 164, in build_sources\n",
      "      File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/command/build_src.py\", line 164, in build_sources\n",
      "      File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/command/build_src.py\", line 323, in build_extension_sources\n",
      "      File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/command/build_src.py\", line 323, in build_extension_sources\n",
      "      File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/command/build_src.py\", line 376, in generate_sources\n",
      "      File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/command/build_src.py\", line 376, in generate_sources\n",
      "      File \"numpy/core/setup.py\", line 407, in generate_config_h\n",
      "      File \"numpy/core/setup.py\", line 407, in generate_config_h\n",
      "        # Version number is added to metadata inside configuration() if build\n",
      "        # Version number is added to metadata inside configuration() if build\n",
      "      File \"numpy/core/setup.py\", line 40, in check_types\n",
      "      File \"numpy/core/setup.py\", line 40, in check_types\n",
      "        License :: OSI Approved :: BSD License\n",
      "        License :: OSI Approved :: BSD License\n",
      "      File \"numpy/core/setup.py\", line 269, in check_types\n",
      "      File \"numpy/core/setup.py\", line 269, in check_types\n",
      "        into a bug, please report it at https://github.com/scipy/scipy/issues.\n",
      "        into a bug, please report it at https://github.com/scipy/scipy/issues.\n",
      "    SystemError: Cannot compile 'Python.h'. Perhaps you need to install python-dev|python-devel.\n",
      "    SystemError: Cannot compile 'Python.h'. Perhaps you need to install python-dev|python-devel.\n",
      "    Complete output from command python setup.py egg_info:\n",
      "    Complete output from command python setup.py egg_info:\n",
      "    Running from numpy source directory.\n",
      "\n",
      "/tmp/easy_install-3_H80f/numpy-1.11.2rc1/setup.py:327: UserWarning: Unrecognized setuptools command, proceeding with generating Cython sources and expanding templates\n",
      "\n",
      "  warnings.warn(\"Unrecognized setuptools command, proceeding with \"\n",
      "\n",
      "/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/system_info.py:1630: UserWarning:\n",
      "\n",
      "    Atlas (http://math-atlas.sourceforge.net/) libraries not found.\n",
      "\n",
      "    Directories to search for the libraries can be specified in the\n",
      "\n",
      "    numpy/distutils/site.cfg file (section [atlas]) or by setting\n",
      "\n",
      "    the ATLAS environment variable.\n",
      "\n",
      "  warnings.warn(AtlasNotFoundError.__doc__)\n",
      "\n",
      "/usr/bin/ld: cannot find -lcblas\n",
      "\n",
      "collect2: error: ld returned 1 exit status\n",
      "\n",
      "/bin/sh: 1: svnversion: not found\n",
      "\n",
      "/bin/sh: 1: svnversion: not found\n",
      "\n",
      "/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/system_info.py:1532: UserWarning:\n",
      "\n",
      "    Atlas (http://math-atlas.sourceforge.net/) libraries not found.\n",
      "\n",
      "    Directories to search for the libraries can be specified in the\n",
      "\n",
      "    numpy/distutils/site.cfg file (section [atlas]) or by setting\n",
      "\n",
      "    the ATLAS environment variable.\n",
      "\n",
      "  warnings.warn(AtlasNotFoundError.__doc__)\n",
      "\n",
      "/usr/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'define_macros'\n",
      "\n",
      "  warnings.warn(msg)\n",
      "\n",
      "_configtest.c:1:5: warning: conflicting types for built-in function ‘exp’ [enabled by default]\n",
      "\n",
      " int exp (void);\n",
      "\n",
      "     ^\n",
      "\n",
      "_configtest.o: In function `main':\n",
      "\n",
      "/tmp/easy_install-3_H80f/numpy-1.11.2rc1/_configtest.c:6: undefined reference to `exp'\n",
      "\n",
      "collect2: error: ld returned 1 exit status\n",
      "\n",
      "_configtest.c:1:5: warning: conflicting types for built-in function ‘exp’ [enabled by default]\n",
      "\n",
      " int exp (void);\n",
      "\n",
      "     ^\n",
      "\n",
      "_configtest.c:1:20: fatal error: Python.h: No such file or directory\n",
      "\n",
      " #include <Python.h>\n",
      "\n",
      "                    ^\n",
      "\n",
      "compilation terminated.\n",
      "\n",
      "/usr/bin/ld: cannot find -lcblas\n",
      "\n",
      "collect2: error: ld returned 1 exit status\n",
      "\n",
      "non-existing path in 'numpy/distutils': 'site.cfg'\n",
      "\n",
      "_configtest.o: In function `main':\n",
      "\n",
      "/tmp/easy_install-3_H80f/numpy-1.11.2rc1/_configtest.c:6: undefined reference to `exp'\n",
      "\n",
      "collect2: error: ld returned 1 exit status\n",
      "\n",
      "_configtest.c:1:20: fatal error: Python.h: No such file or directory\n",
      "\n",
      " #include <Python.h>\n",
      "\n",
      "                    ^\n",
      "\n",
      "compilation terminated.\n",
      "\n",
      "Traceback (most recent call last):\n",
      "\n",
      "  File \"<string>\", line 17, in <module>\n",
      "\n",
      "  File \"/tmp/pip_build_root/scipy/setup.py\", line 415, in <module>\n",
      "\n",
      "    setup_package()\n",
      "\n",
      "  File \"/tmp/pip_build_root/scipy/setup.py\", line 411, in setup_package\n",
      "\n",
      "    setup(**metadata)\n",
      "\n",
      "  File \"/usr/lib/python2.7/distutils/core.py\", line 111, in setup\n",
      "\n",
      "    _setup_distribution = dist = klass(attrs)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/dist.py\", line 315, in __init__\n",
      "\n",
      "    self.fetch_build_eggs(attrs['setup_requires'])\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/dist.py\", line 361, in fetch_build_eggs\n",
      "\n",
      "    replace_conflicting=True,\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py\", line 851, in resolve\n",
      "\n",
      "    dist = best[req.key] = env.best_match(req, ws, installer)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py\", line 1123, in best_match\n",
      "\n",
      "    return self.obtain(req, installer)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py\", line 1135, in obtain\n",
      "\n",
      "    return installer(requirement)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/dist.py\", line 429, in fetch_build_egg\n",
      "\n",
      "    return cmd.easy_install(req)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/easy_install.py\", line 665, in easy_install\n",
      "\n",
      "    return self.install_item(spec, dist.location, tmpdir, deps)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/easy_install.py\", line 695, in install_item\n",
      "\n",
      "    dists = self.install_eggs(spec, download, tmpdir)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/easy_install.py\", line 876, in install_eggs\n",
      "\n",
      "    return self.build_and_install(setup_script, setup_base)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/easy_install.py\", line 1115, in build_and_install\n",
      "\n",
      "    self.run_setup(setup_script, setup_base, args)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/easy_install.py\", line 1101, in run_setup\n",
      "\n",
      "    run_setup(setup_script, args)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 249, in run_setup\n",
      "\n",
      "    raise\n",
      "\n",
      "  File \"/usr/lib/python2.7/contextlib.py\", line 35, in __exit__\n",
      "\n",
      "    self.gen.throw(type, value, traceback)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 197, in setup_context\n",
      "\n",
      "    yield\n",
      "\n",
      "  File \"/usr/lib/python2.7/contextlib.py\", line 35, in __exit__\n",
      "\n",
      "    self.gen.throw(type, value, traceback)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 168, in save_modules\n",
      "\n",
      "    saved_exc.resume()\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 143, in resume\n",
      "\n",
      "    six.reraise(type, exc, self._tb)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 156, in save_modules\n",
      "\n",
      "    yield saved\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 197, in setup_context\n",
      "\n",
      "    yield\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 246, in run_setup\n",
      "\n",
      "    DirectorySandbox(setup_dir).run(runner)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 276, in run\n",
      "\n",
      "    return func()\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 245, in runner\n",
      "\n",
      "    _execfile(setup_script, ns)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 47, in _execfile\n",
      "\n",
      "    exec(code, globals, locals)\n",
      "\n",
      "  File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/setup.py\", line 386, in <module>\n",
      "\n",
      "    # Raise errors for unsupported commands, improve help output, etc.\n",
      "\n",
      "  File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/setup.py\", line 378, in setup_package\n",
      "\n",
      "    test_suite='nose.collector',\n",
      "\n",
      "  File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/core.py\", line 169, in setup\n",
      "\n",
      "  File \"/usr/lib/python2.7/distutils/core.py\", line 151, in setup\n",
      "\n",
      "    dist.run_commands()\n",
      "\n",
      "  File \"/usr/lib/python2.7/distutils/dist.py\", line 953, in run_commands\n",
      "\n",
      "    self.run_command(cmd)\n",
      "\n",
      "  File \"/usr/lib/python2.7/distutils/dist.py\", line 972, in run_command\n",
      "\n",
      "    cmd_obj.run()\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/bdist_egg.py\", line 152, in run\n",
      "\n",
      "    self.run_command(\"egg_info\")\n",
      "\n",
      "  File \"/usr/lib/python2.7/distutils/cmd.py\", line 326, in run_command\n",
      "\n",
      "    self.distribution.run_command(command)\n",
      "\n",
      "  File \"/usr/lib/python2.7/distutils/dist.py\", line 972, in run_command\n",
      "\n",
      "    cmd_obj.run()\n",
      "\n",
      "  File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/command/egg_info.py\", line 18, in run\n",
      "\n",
      "  File \"/usr/lib/python2.7/distutils/cmd.py\", line 326, in run_command\n",
      "\n",
      "    self.distribution.run_command(command)\n",
      "\n",
      "  File \"/usr/lib/python2.7/distutils/dist.py\", line 972, in run_command\n",
      "\n",
      "    cmd_obj.run()\n",
      "\n",
      "  File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/command/build_src.py\", line 147, in run\n",
      "\n",
      "  File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/command/build_src.py\", line 164, in build_sources\n",
      "\n",
      "  File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/command/build_src.py\", line 323, in build_extension_sources\n",
      "\n",
      "  File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/command/build_src.py\", line 376, in generate_sources\n",
      "\n",
      "  File \"numpy/core/setup.py\", line 407, in generate_config_h\n",
      "\n",
      "    # Version number is added to metadata inside configuration() if build\n",
      "\n",
      "  File \"numpy/core/setup.py\", line 40, in check_types\n",
      "\n",
      "    License :: OSI Approved :: BSD License\n",
      "\n",
      "  File \"numpy/core/setup.py\", line 269, in check_types\n",
      "\n",
      "    into a bug, please report it at https://github.com/scipy/scipy/issues.\n",
      "\n",
      "SystemError: Cannot compile 'Python.h'. Perhaps you need to install python-dev|python-devel.\n",
      "\n",
      "----------------------------------------\n",
      "    Running from numpy source directory.\n",
      "\n",
      "/tmp/easy_install-3_H80f/numpy-1.11.2rc1/setup.py:327: UserWarning: Unrecognized setuptools command, proceeding with generating Cython sources and expanding templates\n",
      "\n",
      "  warnings.warn(\"Unrecognized setuptools command, proceeding with \"\n",
      "\n",
      "/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/system_info.py:1630: UserWarning:\n",
      "\n",
      "    Atlas (http://math-atlas.sourceforge.net/) libraries not found.\n",
      "\n",
      "    Directories to search for the libraries can be specified in the\n",
      "\n",
      "    numpy/distutils/site.cfg file (section [atlas]) or by setting\n",
      "\n",
      "    the ATLAS environment variable.\n",
      "\n",
      "  warnings.warn(AtlasNotFoundError.__doc__)\n",
      "\n",
      "/usr/bin/ld: cannot find -lcblas\n",
      "\n",
      "collect2: error: ld returned 1 exit status\n",
      "\n",
      "/bin/sh: 1: svnversion: not found\n",
      "\n",
      "/bin/sh: 1: svnversion: not found\n",
      "\n",
      "/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/system_info.py:1532: UserWarning:\n",
      "\n",
      "    Atlas (http://math-atlas.sourceforge.net/) libraries not found.\n",
      "\n",
      "    Directories to search for the libraries can be specified in the\n",
      "\n",
      "    numpy/distutils/site.cfg file (section [atlas]) or by setting\n",
      "\n",
      "    the ATLAS environment variable.\n",
      "\n",
      "  warnings.warn(AtlasNotFoundError.__doc__)\n",
      "\n",
      "/usr/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'define_macros'\n",
      "\n",
      "  warnings.warn(msg)\n",
      "\n",
      "_configtest.c:1:5: warning: conflicting types for built-in function ‘exp’ [enabled by default]\n",
      "\n",
      " int exp (void);\n",
      "\n",
      "     ^\n",
      "\n",
      "_configtest.o: In function `main':\n",
      "\n",
      "/tmp/easy_install-3_H80f/numpy-1.11.2rc1/_configtest.c:6: undefined reference to `exp'\n",
      "\n",
      "collect2: error: ld returned 1 exit status\n",
      "\n",
      "_configtest.c:1:5: warning: conflicting types for built-in function ‘exp’ [enabled by default]\n",
      "\n",
      " int exp (void);\n",
      "\n",
      "     ^\n",
      "\n",
      "_configtest.c:1:20: fatal error: Python.h: No such file or directory\n",
      "\n",
      " #include <Python.h>\n",
      "\n",
      "                    ^\n",
      "\n",
      "compilation terminated.\n",
      "\n",
      "/usr/bin/ld: cannot find -lcblas\n",
      "\n",
      "collect2: error: ld returned 1 exit status\n",
      "\n",
      "non-existing path in 'numpy/distutils': 'site.cfg'\n",
      "\n",
      "_configtest.o: In function `main':\n",
      "\n",
      "/tmp/easy_install-3_H80f/numpy-1.11.2rc1/_configtest.c:6: undefined reference to `exp'\n",
      "\n",
      "collect2: error: ld returned 1 exit status\n",
      "\n",
      "_configtest.c:1:20: fatal error: Python.h: No such file or directory\n",
      "\n",
      " #include <Python.h>\n",
      "\n",
      "                    ^\n",
      "\n",
      "compilation terminated.\n",
      "\n",
      "Traceback (most recent call last):\n",
      "\n",
      "  File \"<string>\", line 17, in <module>\n",
      "\n",
      "  File \"/tmp/pip_build_root/scipy/setup.py\", line 415, in <module>\n",
      "\n",
      "    setup_package()\n",
      "\n",
      "  File \"/tmp/pip_build_root/scipy/setup.py\", line 411, in setup_package\n",
      "\n",
      "    setup(**metadata)\n",
      "\n",
      "  File \"/usr/lib/python2.7/distutils/core.py\", line 111, in setup\n",
      "\n",
      "    _setup_distribution = dist = klass(attrs)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/dist.py\", line 315, in __init__\n",
      "\n",
      "    self.fetch_build_eggs(attrs['setup_requires'])\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/dist.py\", line 361, in fetch_build_eggs\n",
      "\n",
      "    replace_conflicting=True,\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py\", line 851, in resolve\n",
      "\n",
      "    dist = best[req.key] = env.best_match(req, ws, installer)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py\", line 1123, in best_match\n",
      "\n",
      "    return self.obtain(req, installer)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/pkg_resources/__init__.py\", line 1135, in obtain\n",
      "\n",
      "    return installer(requirement)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/dist.py\", line 429, in fetch_build_egg\n",
      "\n",
      "    return cmd.easy_install(req)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/easy_install.py\", line 665, in easy_install\n",
      "\n",
      "    return self.install_item(spec, dist.location, tmpdir, deps)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/easy_install.py\", line 695, in install_item\n",
      "\n",
      "    dists = self.install_eggs(spec, download, tmpdir)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/easy_install.py\", line 876, in install_eggs\n",
      "\n",
      "    return self.build_and_install(setup_script, setup_base)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/easy_install.py\", line 1115, in build_and_install\n",
      "\n",
      "    self.run_setup(setup_script, setup_base, args)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/easy_install.py\", line 1101, in run_setup\n",
      "\n",
      "    run_setup(setup_script, args)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 249, in run_setup\n",
      "\n",
      "    raise\n",
      "\n",
      "  File \"/usr/lib/python2.7/contextlib.py\", line 35, in __exit__\n",
      "\n",
      "    self.gen.throw(type, value, traceback)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 197, in setup_context\n",
      "\n",
      "    yield\n",
      "\n",
      "  File \"/usr/lib/python2.7/contextlib.py\", line 35, in __exit__\n",
      "\n",
      "    self.gen.throw(type, value, traceback)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 168, in save_modules\n",
      "\n",
      "    saved_exc.resume()\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 143, in resume\n",
      "\n",
      "    six.reraise(type, exc, self._tb)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 156, in save_modules\n",
      "\n",
      "    yield saved\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 197, in setup_context\n",
      "\n",
      "    yield\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 246, in run_setup\n",
      "\n",
      "    DirectorySandbox(setup_dir).run(runner)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 276, in run\n",
      "\n",
      "    return func()\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 245, in runner\n",
      "\n",
      "    _execfile(setup_script, ns)\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/sandbox.py\", line 47, in _execfile\n",
      "\n",
      "    exec(code, globals, locals)\n",
      "\n",
      "  File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/setup.py\", line 386, in <module>\n",
      "\n",
      "    # Raise errors for unsupported commands, improve help output, etc.\n",
      "\n",
      "  File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/setup.py\", line 378, in setup_package\n",
      "\n",
      "    test_suite='nose.collector',\n",
      "\n",
      "  File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/core.py\", line 169, in setup\n",
      "\n",
      "  File \"/usr/lib/python2.7/distutils/core.py\", line 151, in setup\n",
      "\n",
      "    dist.run_commands()\n",
      "\n",
      "  File \"/usr/lib/python2.7/distutils/dist.py\", line 953, in run_commands\n",
      "\n",
      "    self.run_command(cmd)\n",
      "\n",
      "  File \"/usr/lib/python2.7/distutils/dist.py\", line 972, in run_command\n",
      "\n",
      "    cmd_obj.run()\n",
      "\n",
      "  File \"/usr/local/lib/python2.7/dist-packages/setuptools/command/bdist_egg.py\", line 152, in run\n",
      "\n",
      "    self.run_command(\"egg_info\")\n",
      "\n",
      "  File \"/usr/lib/python2.7/distutils/cmd.py\", line 326, in run_command\n",
      "\n",
      "    self.distribution.run_command(command)\n",
      "\n",
      "  File \"/usr/lib/python2.7/distutils/dist.py\", line 972, in run_command\n",
      "\n",
      "    cmd_obj.run()\n",
      "\n",
      "  File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/command/egg_info.py\", line 18, in run\n",
      "\n",
      "  File \"/usr/lib/python2.7/distutils/cmd.py\", line 326, in run_command\n",
      "\n",
      "    self.distribution.run_command(command)\n",
      "\n",
      "  File \"/usr/lib/python2.7/distutils/dist.py\", line 972, in run_command\n",
      "\n",
      "    cmd_obj.run()\n",
      "\n",
      "  File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/command/build_src.py\", line 147, in run\n",
      "\n",
      "  File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/command/build_src.py\", line 164, in build_sources\n",
      "\n",
      "  File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/command/build_src.py\", line 323, in build_extension_sources\n",
      "\n",
      "  File \"/tmp/easy_install-3_H80f/numpy-1.11.2rc1/numpy/distutils/command/build_src.py\", line 376, in generate_sources\n",
      "\n",
      "  File \"numpy/core/setup.py\", line 407, in generate_config_h\n",
      "\n",
      "    # Version number is added to metadata inside configuration() if build\n",
      "\n",
      "  File \"numpy/core/setup.py\", line 40, in check_types\n",
      "\n",
      "    License :: OSI Approved :: BSD License\n",
      "\n",
      "  File \"numpy/core/setup.py\", line 269, in check_types\n",
      "\n",
      "    into a bug, please report it at https://github.com/scipy/scipy/issues.\n",
      "\n",
      "SystemError: Cannot compile 'Python.h'. Perhaps you need to install python-dev|python-devel.\n",
      "\n",
      "----------------------------------------\n",
      "Cleaning up...\n",
      "Cleaning up...\n",
      "Command python setup.py egg_info failed with error code 1 in /tmp/pip_build_root/scipy\n",
      "Command python setup.py egg_info failed with error code 1 in /tmp/pip_build_root/scipy\n"
     ]
    },
    {
     "ename": "UnicodeDecodeError",
     "evalue": "'ascii' codec can't decode byte 0xe2 in position 72: ordinal not in range(128)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m\u001b[0m",
      "\u001b[0;31mUnicodeDecodeError\u001b[0mTraceback (most recent call last)",
      "\u001b[0;32m<ipython-input-2-a27a186c18dc>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m     11\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m \u001b[0;31m# Install packages\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 13\u001b[0;31m \u001b[0minstall\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpackages\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m<ipython-input-2-a27a186c18dc>\u001b[0m in \u001b[0;36minstall\u001b[0;34m(packages)\u001b[0m\n\u001b[1;32m      8\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0minstall\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mpackages\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      9\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mpackage\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mpackages\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 10\u001b[0;31m         \u001b[0mpip\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmain\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'install'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mpackage\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     11\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m \u001b[0;31m# Install packages\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python2.7/dist-packages/pip/__init__.pyc\u001b[0m in \u001b[0;36mmain\u001b[0;34m(initial_args)\u001b[0m\n\u001b[1;32m    233\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    234\u001b[0m     \u001b[0mcommand\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcommands\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mcmd_name\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 235\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mcommand\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmain\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcmd_args\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    236\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    237\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib/python2.7/dist-packages/pip/basecommand.pyc\u001b[0m in \u001b[0;36mmain\u001b[0;34m(self, args)\u001b[0m\n\u001b[1;32m    159\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mstore_log\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    160\u001b[0m             \u001b[0mlog_file_fn\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0moptions\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlog_file\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 161\u001b[0;31m             \u001b[0mtext\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m'\\n'\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcomplete_log\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    162\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    163\u001b[0m                 \u001b[0mlog_file_fp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mopen_logfile\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlog_file_fn\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'w'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mUnicodeDecodeError\u001b[0m: 'ascii' codec can't decode byte 0xe2 in position 72: ordinal not in range(128)"
     ]
    }
   ],
   "source": [
    "# Import pip\n",
    "import pip\n",
    "\n",
    "# Define packages to install\n",
    "packages = ['scipy']\n",
    "\n",
    "# Define install function\n",
    "def install(packages):\n",
    "    for package in packages:\n",
    "        pip.main(['install', package])\n",
    "        \n",
    "# Install packages\n",
    "install(packages)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
